R/pkg/R/DataFrame.R

```diff
@@ -3250,6 +3250,7 @@ setMethod("attach",
           function(what, pos = 2, name = deparse(substitute(what)), warn.conflicts = TRUE) {
             newEnv <- assignNewEnv(what)
             attach(newEnv, pos = pos, name = name, warn.conflicts = warn.conflicts)
+            on.exit(detach(newEnv, pos = pos, name = name))
```

Comment:
The point of this function is to allow users to attach the column names of the SparkDataFrame into their environment. So in some sense it is mimic'ing `attach(df)` in R. Following this users can directly use the column name in expressions like `select` etc.

So adding a detach on function exit defeats the purpose. The only way I see to not use attach is to remove this feature ?

cc @felixcheung

Reply:
The reported point is a NOTE, as per my understanding, we should be able to release with a NOTE (in fact, there is another NOTE on an unknown maintainer...)

Also, I'm not sure how by addition this would address the NOTE? From what I can see, the NOTE says we should be careful about having an `attach` method... for which I see the only way to address it or to avoid the NOTE is... to remove the `attach` method?